SQL Fix Closes #104

gmross opened the pull request:
These are the fixes that I managed to pull together based on Owen's comments in #104 (SQL Fixes).  I believe this closes the issue(s) addressed.

Commit:
Merge pull request #1 from clemsonMakerspace/devel

Test

Commit:
Update requirements.txt

Commit:
Merge remote-tracking branch 'refs/remotes/clemsonMakerspace/devel' into devel

gmross committed:
Update 0.1?

Not really sure

gmross committed:
Update ErrorSQL.py

gmross committed:
Update main.py

gmross committed:
Update AuthenticationSQL.py

gmross committed:
Merge pull request #2 from gmross/gmross-patch-1

Update 0.1?

ophilli committed:
Merge branch 'devel' into devel

ophilli commented on Software/scan_key/AuthenticationSQL.py:
```diff
@@ -1,42 +1,30 @@
+    cursor.execute("SELECT auth1, auth2, auth3 FROM USER WHERE t1String = " + id) #getting user w/ the id    #might need to modify
 
-    machineType = None
+    authdata = cursor.fetchone()
```
We don't want to download all three cols - just the one that is appropriate for our machine type.

ophilli commented on Software/scan_key/AuthenticationSQL.py:
```diff
@@ -1,42 +1,30 @@
-    authdata = cursor.fetchall()
-
-    for auth in authdata:
-        if(auth[machineType] != 1):
```
The new line 17 should replace this logic.

r3cebarnett committed:
Merge branch 'devel' into devel

r3cebarnett committed:
Update AuthenticationSQL.py

Okay so I fixed how we get the stuffs from the SQL. There were a few things that were pretty screwy. So now we look up "authType" using "PiHostname" in the SQL, then look up the variable that authType is for ID in user. This is what is returned in machineAuth. For some reason, we had machineAuth returning the current machine, which makes no sense as we try to use it as a boolean later in the program but also as a machinetype in the SQL call in line 54, 55. This is like really stupid and needs to be fixed but I'm more concerned about fixing the algorithm than an event log. @gacosta, @gmross, and I will discuss a better way for us to do this. Also I'd like us to clear up a lot of the formatting for our code. It's very sporadic and all over the place where functions are. It'd be a lot of cleaning to do but I'd really like us to reorganize where we put functions. I found myself searching for a long time trying to find a specific function.

Anyways /rant. @ophilli does this fix the issue you had or do you have any further comments?

r3cebarnett commented:
Everything that I did can be seen in commit# 42577a7

Okay so I fixed how we get the stuffs from the SQL. There were a few things that were pretty screwy. So now we look up "authType" using "PiHostname" in the SQL, then look up the variable that authType is for ID in user. This is what is returned in machineAuth. For some reason, we had machineAuth returning the current machine, which makes no sense as we try to use it as a boolean later in the program but also as a machinetype in the SQL call in line 54, 55. This is like really stupid and needs to be fixed but I'm more concerned about fixing the algorithm than an event log. @gacosta, @gmross, and I will discuss a better way for us to do this. Also I'd like us to clear up a lot of the formatting for our code. It's very sporadic and all over the place where functions are. It'd be a lot of cleaning to do but I'd really like us to reorganize where we put functions. I found myself searching for a long time trying to find a specific function.

Anyways /rant. @ophilli does this fix the issue you had or do you have any further comments?

r3cebarnett committed:
Update AuthenticationSQL.py

ophilli commented on Software/scan_key/AuthenticationSQL.py:
```diff
@@ -8,16 +8,16 @@ def machineAuth(id, cursor):
+    authType = cursor.fetchone()
 
-    cursor.execute("SELECT auth1, auth2, auth3 FROM USER WHERE t1String = " + id) #getting user w/ the id    #might need to modify
+    cursor.execute("SELECT " + authType + " FROM USER WHERE t1String = " + id) #getting user w/ the id    #might need to modify
```
t1String is not a column in the `catsadmin`.`users` table on watt.cuMaker.space.

Will accept once this issue is resolved. Might be resolved by adjusting code, or simply adding the column to the table. I'll leave this up to the software team to review.

ophilli reviewed (changes requested):
Your changes look fine to me - except for the t1String issue.
